src/backlight/strategies/exit.py

```diff
@@ -125,11 +125,29 @@ def _exit_by_max_holding_time(
                 continue
 
             idx = trade.index[0]
-            df_exit = df[(idx <= df.index) & (df.index <= idx + max_holding_time)]
-            transaction = _exit_transaction(df_exit, trade, exit_condition)
-
-            indices.append(transaction.timestamp)
-            exits.append((transaction.amount, i))
+            #             df_exit = df[(idx <= df.index) & (df.index <= idx + max_holding_time)]
+            #             transaction = _exit_transaction(df_exit, trade, exit_condition)
+
+            #             indices.append(transaction.timestamp)
+            #             exits.append((transaction.amount, i))
+
+            freq = df.index[1] - df.index[0]
+            start = max(idx, df.index[0])
+            end = min(idx + max_holding_time, df.index[-1])
+
+            for indx in pd.date_range(start=start, end=end, freq=freq):
```

Comment:
you need not to create new index, because we can not garantee that df.index freq is `df.index[1] - df.index[0]`.
Just use df.index s.t

```
for ind in df.index:
```

Reply:
In my understanding, it is better to write these changes in `_exit_transaction`?